ui/components/app/multilayer-fee-message/multi-layer-fee-message.js

```diff
@@ -0,0 +1,70 @@
+import React, { useContext, useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { captureException } from '@sentry/browser';
+import TransactionDetailItem from '../transaction-detail-item/transaction-detail-item.component';
+import fetchEstimatedL1Fee from '../../../helpers/utils/optimism/fetchEstimatedL1Fee';
+import { I18nContext } from '../../../contexts/i18n';
+import { sumHexes } from '../../../helpers/utils/transactions.util';
+import {
+  toBigNumber,
+  toNormalizedDenomination,
+} from '../../../../shared/modules/conversion.utils';
+
+export default function MultilayerFeeMessage({ transaction, layer2fee }) {
```

Comment:
Ah, you decided to make a separate component for this. I like this implementation, seems easy to see what's going on.